packages/smart-wallet/src/marshal-contexts.js

```diff
@@ -9,7 +9,9 @@ const { Fail, quote: q } = assert;
  * For a value with a known id in the board, we can use
  * that board id as a slot to preserve identity when marshaling.
  *
- * @typedef {`board${Digits}`} BoardId
+ *  typedef {`board${Digits}`} BoardId // this was useful for development
+ *
+ * @typedef {string} BoardId
```

Comment:
this should work ([playground](https://www.typescriptlang.org/play?#code/C4TwDgpgBAQg9gQwE4BMCSKoF4oAMBGiqAJAN4B2ArgLb4RIC+uA3AFCsA2EwUhy6KAFywiAtqz6oM2KAHJJKAIyzxC6TnmiAYnDgr2azBoWKADOfP6Jo9XIWn9hmZv4BaRQCYAzFafHR+kA))
```suggestion
 *  @typedef {`board${number}`} BoardId
```

more on [TS literal types](https://www.typescriptlang.org/docs/handbook/2/template-literal-types.html)

Reply:
Well, it's better, but we don't statically fix the prefix either. And literals such as `'board0374'` used in tests have type `string`. Is it cost-effective to cast them? I sort of doubt that users of our API would thank us for that sort of thing. But I'm open to discussion.

Reply:
> cost-effective to cast them

I suppose not. If anyone ever confuses a non-boardId string with a boardId then we can revisit :)